docs/writing-tests/testharness.md

```diff
@@ -1,114 +1,93 @@
 # JavaScript Tests (testharness.js)
 
-```eval_rst
-.. toctree::
-   :maxdepth: 1
-
-   idlharness
-   testharness-api
-   testdriver-extension-tutorial
-   testdriver
-```
-
-testharness.js tests are the correct type of test to write in any
+JavaScript tests are the correct type of test to write in any
 situation where you are not specifically interested in the rendering
 of a page, and where human interaction isn't required; these tests are
-written in JavaScript using a framework called `testharness.js`. It is
-documented in two sections:
+written in JavaScript using a framework called `testharness.js`.
+
+A high-level overview is provided below and more information can be found here:
 
   * [testharness.js Documentation](testharness-api.md) — An introduction
-    to the library and a detailed API reference.
+    to the library and a detailed API reference. [The tutorial on writing a
+    testharness.js test](testharness-tutorial) provides a concise guide to writing
+    a test — a good place to start for newcomers to the project.
+
+  * [testdriver.js Automation](testdriver.md) — Automating end user actions, such as moving or
+    clicking a mouse.
 
   * [idlharness.js Documentation](idlharness.md) — A library for testing
      IDL interfaces using `testharness.js`.
 
 See [server features](server-features.md) for advanced testing features that are commonly used
-with testharness.js. See also the [general guidelines](general-guidelines.md) for all test types.
+with JavaScript tests. See also the [general guidelines](general-guidelines.md) for all test types.
 
-This page describes testharness.js exhaustively; [the tutorial on writing a
-testharness.js test](testharness-tutorial) provides a concise guide to writing
-a test--a good place to start for newcomers to the project.
+## Window tests
```

Comment:
I can see how this might be a more logical structure for test authors! As a tooling/infra maintainer, I always think of it as "I write the test page" vs. "server generates the test page" on the top level.

Reply:
Yeah, server documentation is still suboptimal and might warrant another look to ensure common patterns are covered adequately.